src/cascadia/TerminalSettingsModel/DefaultProfileUtils.cpp

```diff
@@ -20,14 +20,15 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CreateDefaultProfile(const
     const winrt::guid profileGuid{ Microsoft::Console::Utils::CreateV5Uuid(TERMINAL_PROFILE_NAMESPACE_GUID,
                                                                            gsl::as_bytes(gsl::make_span(name))) };
 
-    auto newProfile = winrt::make<winrt::Microsoft::Terminal::Settings::Model::implementation::Profile>(profileGuid);
-    newProfile.Name(name);
+    auto newProfile = winrt::make_self<winrt::Microsoft::Terminal::Settings::Model::implementation::Profile>(profileGuid);
+    newProfile->Name(winrt::hstring{ name });

```

Comment:
is there a reason we need the explicit `hstring` constructor here, when we didn't before?

Reply:
Yeah, for some reason, we get this error:
```
8>D:\Terminal\src\cascadia\TerminalSettingsModel\DefaultProfileUtils.cpp(24,26): error C2664: 'void winrt::Microsoft::Terminal::Settings::Model::implementation::Profile::Name(const winrt::hstring &)': cannot convert argument 1 from 'const std::wstring_view' to 'const winrt::hstring &'
8>D:\Terminal\src\cascadia\TerminalSettingsModel\DefaultProfileUtils.cpp(24,26): message : Reason: cannot convert from 'const std::wstring_view' to 'const winrt::hstring'
8>D:\Terminal\src\cascadia\TerminalSettingsModel\DefaultProfileUtils.cpp(24,22): message : No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
8>D:\Terminal\src\cascadia\TerminalSettingsModel\Profile.h(67,9): message : see declaration of 'winrt::Microsoft::Terminal::Settings::Model::implementation::Profile::Name' (compiling source file DefaultProfileUtils.cpp)
```

Figured the explicit cast would be better than `.c_str()`

Reply:
This is one of those weird c++/winrt "do as I say, not as I do" things.

Generated projection types use `winrt::param::hstring` as their string type, which has an implicit conversion from `wstring_view` to allow you to pass a string view directly.

`winrt::hstring` does not.

Just FYI, though: _this is not a cast_. You are constructing a new hstring temporary by invoking its constructor using universal initializer syntax.

Reply:
Oh I forgot about the `param::hstring` thing.